### PR TITLE
Add load balancing notes to dev guide

### PR DESCRIPTION
**Pull request by moxcodes**
## Proposed changes

Co-Authored with @fmahebert

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

**Review by nilsvu — DISMISSED** [review]
Thanks for writing this up 👍

**Comment by nikwit** [comment]
do you actually have to specify the event/trigger for load balancing in the executable or is this done automatically?

**Comment by moxcodes** [comment]
The load balancing has to be present in the `phase_changes` typelist in the metavariables, and the executable needs to satisfy a few interface requirements for supporting the phase changing infrastructure -- it's similar to, but separate from the events and triggers.
Currently, the set of alterations that need to be made for the phase changes isn't documented thoroughly in a centralized place, but the documentation for `PhaseChanges`, `PhaseControl` and `PhaseChangeAndTriggers` offers some of the technical details, and there are examples of existing executables that have the feature, e.g. ScalarWave.

**Comment by nilsvu** [comment]
@moxcodes Do you think you'll be able to do #3266, i.e. put the phase-control related metavariables in a struct and add some docs to it? If you find the time, that would be very valuable for future maintenance I think.

**Comment by moxcodes** [comment]
Yes, I'll try to find the time to add that struct and documentation.

File: docs/DevGuide/LoadBalancingNotes.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    - - VisitAndReturn(LoadBalancing)
+```
+- Use `LBTurnInstrumentOff` and `LBTurnInstrumentOn` to specifically exclude
+  setup procedures from the LB instrumentation. First attempts indicate that
+  this process might be challenging to accomplish correctly, and may require
+  correspondence with the Charm developers to clarify at what points in the
+  code execution those commands may be used, and precisely how they affect
+  the load-balancing database. A first attempt by JM was to turn instrumentation
+  off during array element construction, then turn instrumentation on for each
+  element during the start of the `Evolve` phase, but that attempt lead to a
```
Comment (kidder) [inline docs/DevGuide/LoadBalancingNotes.md]: lead -> led

**Review by kidder — CHANGES_REQUESTED** [review]
Thank you for writing this up; you can squash the one typo

**Comment by moxcodes** [comment]
squashed in the typo fix.
Cheers!

**Review by kidder — APPROVED** [review]

**Pull request merged**